src/vs/workbench/parts/terminal/electron-browser/terminalService.ts

```diff
@@ -80,6 +80,7 @@ export class TerminalService implements ITerminalService {
 			this.setActiveInstanceByIndex(0);
 		}
 		this._onInstancesChanged.fire();
+		terminalInstance.setCommandsToSkipShell(this.configHelper.getCommandsToSkipShell());
```

Comment:
You've actually stumbled upon a bug bigger than this in that no settings values are updated correctly, including scrollback and cursorBlinking as well. Having a brief look at the code, I think the best way to fix this might be to move `TerminalPanel._updateConfig` to `TerminalService.updateConfig` and then call into that after creating the new instance.